src/library/Tabs/Tabs.js

```diff
@@ -0,0 +1,295 @@
+import Tab from './Tab';
+import TabList from './TabList';
+import TabPanel from './TabPanel';
+
+type Props = {
+  /**  Horizontal or vertical alignment of Tabs in the tab list */
+  align?: 'start' | 'center' | 'end' | 'justify',
+  /** @Private */
+  'aria-labelledby'?: string,
+  /** Content of Tabs; must be Tab components */
+  children?: React$Node,
+  /**
+   * Index of the selected Tab; primarily for use with uncontrolled components
+   */
+  defaultSelectedTabIndex?: number,
+  /** Id of Tabs */
+  id?: string,
+  /** Accessible label for Tabs */
+  label: string,
+  /** Maximum height of Tabs */
+  maxHeight?: number | string,
+  /** Maximum width of each Tab */
+  maxTabWidth?: number | string,
+  /** Called when a Tab is selected */
+  onChange?: (
+    selectedTabIndex: number,
+    event: SyntheticEvent<HTMLAnchorElement>
+  ) => void,
+  /** Position of the tab list in relation to the tab panel */
+  position?: 'bottom' | 'end' | 'start' | 'top',
+  /**
+   * Index of the selected Tab; primarily for use with controlled components.
+   * If this prop is specified, an `onChange` handler must also be specified.
+   * See also: `defaultSelectedTabIndex`
+   */
+  selectedTabIndex?: number,
+  /** @Private Theme */
+  theme: Object
+};
+
+type State = {
+  selectedTabIndex: number
+};
+
+type AnchorEvent = SyntheticEvent<HTMLAnchorElement>;
+
+const Root = createStyledComponent(
+  'div',
+  ({ maxHeight, position }) => {
+    const flexDirection = {
+      bottom: 'column-reverse',
+      end: 'row-reverse',
+      start: 'row',
+      top: 'column'
+    };
+
+    return {
+      display: 'flex',
+      flexDirection: flexDirection[position],
+      maxHeight
+    };
+  },
+  {
+    displayName: 'Tabs',
+    includeStyleReset: true
+  }
+);
+
+/**
+ * Tabs provide easy management for viewing related content in the same layout
+ * region.
+ */
+class Tabs extends Component<Props, State> {
+  static defaultProps = {
+    align: 'start',
+    maxTabWidth: '8.5em',
+    position: 'top'
+  };
+
+  state = {
+    selectedTabIndex: this.props.defaultSelectedTabIndex || 0
+  };
+
+  disabledTabIndexes: Set<number> = new Set();
+
+  id: string = this.props.id || `tabs-${generateId()}`;
+
+  lastIndex: number;
+
+  root: HTMLElement;
+
+  vertical: boolean =
+    this.props.position === 'start' || this.props.position === 'end';
+
+  getPanelId = (index: number) => {
+    return `${this.id}-panel-${index}`;
+  };
+
+  getTabId = (index: number) => {
+    return `${this.id}-tab-${index}`;
+  };
+
+  setRootRef = (node: HTMLElement) => {
+    this.root = node;
+  };
+
+  render() {
+    const {
+      align,
+      children,
+      label,
+      maxHeight,
+      maxTabWidth,
+      position,
+      ...restProps
+    } = this.props;
+    const selectedTabIndex = this.getControllableValue('selectedTabIndex');
+
+    let ids = [];
+    let selectedTabPanelProps;
+    let tabItems = [];
+    const tabs = findAllByType(children, Tab);
```

Comment:
Do we really need to find only Tabs?  I realize that this is all that this component wants, but we could just use children.  I don't think we usually enforce it unless it is necessary for some reason, like in Dialog.